subprojects/kotlin-dsl/src/testFixtures/kotlin/org/gradle/kotlin/dsl/fixtures/ProjectBuilder.kt

```diff
@@ -19,13 +19,28 @@ package org.gradle.kotlin.dsl.fixtures
 import org.gradle.api.Project
 
 import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.testfixtures.internal.ProjectBuilderImpl
 
 import java.io.File
 
 
+/**
+ * Creates a new project using `ProjectBuilder`.
+ *
+ * Make sure to dispose of the project instance by calling [cleanUpProjectBuilderProject].
+ */
 fun newProjectBuilderProjectWith(projectDir: File, name: String = projectDir.name): Project =
```

Comment:
Do you think it would make sense to have a closure taking method here which would automatically close the project after it is finished? So it would look like this:
```
newProjectBuilderProjectWith(...) {
    // do some stuff with project as the receiver
}
```

Or we could have an extension method on `Project` for Kotlin which does this, and you'd use that method instead of `.run`.

Reply:
That's a very good idea, especially given that the only two usages immediately call `.run` on the created instance